core/src/conn.rs

```diff
@@ -160,22 +162,42 @@ impl MongoConnection {
             login_timeout.map(|to| Duration::new(u64::from(to), 0));
         let uuid_repr = user_options.uuid_representation;
         let (client, runtime) = Self::get_client_and_runtime(user_options, runtime)?;
+
+        load_mongosqltranslate_library();
+
+        let type_of_cluster = runtime.block_on(async { determine_cluster_type(&client).await });
+        match type_of_cluster {
+            MongoClusterType::AtlasDataFederation => {}
+            MongoClusterType::Community => {
+                // Community edition is not supported
+                return Err(Error::UnsupportedClusterConfiguration(
+                    "Community edition detected. The driver is intended for use with MongoDB Enterprise edition or Atlas Data Federation.".to_string(),
+                ));
+            }
+            MongoClusterType::Enterprise => {
+                // Ensure the library is loaded if Enterprise edition is detected
+                if get_mongosqltranslate_library().is_none() {
+                    return Err(Error::UnsupportedClusterConfiguration(
+                        "Enterprise edition detected, but mongosqltranslate library not found."
+                            .to_string(),
+                    ));
+                }
+            }
+            MongoClusterType::UnknownTarget => {
+                // Unknown cluster type is not supported
+                return Err(Error::UnsupportedClusterConfiguration(
+                    "Unknown cluster/target type detected. The driver is intended for use with MongoDB Enterprise edition or Atlas Data Federation.".to_string(),
+                ));
+            }
+        }
+
         let connection = MongoConnection {
             client,
             operation_timeout: operation_timeout.map(|to| Duration::new(u64::from(to), 0)),
             uuid_repr,
             runtime,
         };
-        // Verify that the connection is working and the user has access to the default DB
```

Comment:
Not sure if we want to remove this check since it verifies access to the default db which the cluster detection doesn't do.

Reply:
Oh that is a good point. Is that permission check required at connection time according to the ODBC spec, or is it just a nice-to-have to fail earlier? I don't think there's any harm in keeping this, assuming we eventually decide to only support versions of on-prem mongo that support `$documents`.

Reply:
hmm maybe we should keep it then. Is it possible to have access to the admin DB but not the default DB?

Reply:
I looked through the spec and it didn't have guidance on a permission check on connection.  I do think it would be helpful to fail earlier so I'll keep it.  Thanks for the comments on this.